mergify_engine/rules/filter.py

```diff
@@ -115,6 +117,11 @@ class Filter:
         "~=": (lambda a, b: a is not None and b.search(a), any, re.compile),
     }
 
+    multiple_operators: typing.ClassVar[typing.Dict[str, MultipleOperatorT]] = {
+        "or": operator.or_,
+        "and": operator.and_,
```

Comment:
This is the `|` and `&` operators, which is not going to do a lazy evaluation of all conditions.

You want to use `any` and `all` instead.

And you need to make sure each element to `any` and `all` is evaluated with something like:

```
return multiple_op(await element(values) for element in elements)
```

Which will stop (for `any`) as soon as one is true.

Reply:
unfortunately, any() doesn't support async generator :(